trace-mini-agent/src/config.rs

```diff
@@ -28,15 +36,29 @@ impl Config {
             .map_err(|_| anyhow::anyhow!("DD_API_KEY environment variable is not set"))?;
         let mut function_name = None;
 
-        // Google cloud functions automatically sets either K_SERVICE or FUNCTION_NAME
-        // env vars to denote the cloud function name.
-        // K_SERVICE is set on newer runtimes, while FUNCTION_NAME is set on older deprecated runtimes.
+        let mut maybe_env_type = None;
         if let Ok(res) = env::var("K_SERVICE") {
+            // Set by Google Cloud Functions for newer runtimes
             function_name = Some(res);
+            maybe_env_type = Some(trace_utils::EnvironmentType::CloudFunction);
         } else if let Ok(res) = env::var("FUNCTION_NAME") {
+            // Set by Google Cloud Functions for older runtimes
             function_name = Some(res);
+            maybe_env_type = Some(trace_utils::EnvironmentType::CloudFunction);
+        } else if let Ok(res) = env::var("WEBSITE_SITE_NAME") {
+            // Set by Azure Functions
+            function_name = Some(res);
+            maybe_env_type = Some(trace_utils::EnvironmentType::AzureFunction);
         }
 
+        let env_type = maybe_env_type.unwrap_or_else(|| {
```

Comment:
Could we avoid using `#[cfg(not(test))]` and `#[cfg(test)]` here?

Reply:
fixed! process::exit is actually unnecessary here, we can just return